test/test_importer.py

```diff
@@ -1868,39 +1868,37 @@ def mocked_get_release_by_id(
     }
 
     return {
-        "release": {
-            "title": releases[id_][0],
-            "id": id_,
-            "medium-list": [
-                {
-                    "track-list": [
-                        {
-                            "id": "baz",
-                            "recording": {
-                                "title": "foo",
-                                "id": "bar",
-                                "length": 59,
-                            },
-                            "position": 9,
-                            "number": "A2",
-                        }
-                    ],
-                    "position": 5,
-                }
-            ],
-            "artist-credit": [
-                {
-                    "artist": {
-                        "name": releases[id_][1],
-                        "id": "some-id",
-                    },
-                }
-            ],
-            "release-group": {
-                "id": "another-id",
-            },
-            "status": "Official",
-        }
+        "title": releases[id_][0],
+        "id": id_,
+        "media": [
+            {
+                "tracks": [
+                    {
+                        "id": "baz",
+                        "recording": {
+                            "title": "foo",
+                            "id": "bar",
+                            "length": 59,
+                        },
+                        "position": 9,
+                        "number": "A2",
+                    }
+                ],
+                "position": 5,
+            }
+        ],
+        "artist-credit": [
+            {
+                "artist": {
+                    "name": releases[id_][1],
+                    "id": "some-id",
+                },
+            }
+        ],
+        "release-group": {
+            "id": "another-id",
+        },
+        "status": "Official",
```

Comment:
I'll need to do a full review of the PR with these new changes, but there is one thing that could help me get through it a bit faster. Could you explain why the structure of the response dict has changes here? It just raises a red flag when the test fixture has this type of change. 

It only works if everywhere in the code that expected the old structure has changed as well, which I'll need to check and can be time consuming. Because changing the tests without changing the codebase can lead to a mismatch between the tests and the actual code if the fixtures and mocks don't actually relate. Unnesting a value like this can cause big problems.

Reply:
I understand. For a better understanding, you may go commit by commit.

The structure of the response change because the library relies on the json interface instead of the xml as suggested in https://github.com/beetbox/beets/issues/4651#issuecomment-1405310336
The structure of the response is a bit different.

The difference is mainly : 
1. the "release" sublevel has disappeared
2. "xxx-list" has become "xxxs"
3. "medium-list" has become "media"

Reply:
Hello. Eventually, I my split the work and first have a replacement that keeps the xml interface. That would leave the test. The xml is supported by mbzero anyway.
When that's fine, we can think about dropping the xml in favor of json.

Reply:
Ah, thanks! I still need to review, my semester has hit like a truck so I have had very, very little spare time or energy to do code review. I'll get to it soon hopefully :)